hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3224,68 +3307,33 @@ func (r *HostedClusterReconciler) reconcileAWSOIDCDocuments(ctx context.Context,
 	}
 
-	if apiAvailableCondition := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.KubeAPIServerAvailable)); apiAvailableCondition == nil || apiAvailableCondition.Status != metav1.ConditionTrue {
-		return nil
-	}
-	src := &corev1.Secret{
+	secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: hcp.Namespace,
-			Name:      "service-network-admin-kubeconfig",
+			Name:      serviceAccountSigningKeySecret,
 		},
 	}
-	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(src), src); err != nil {
-		return fmt.Errorf("failed to get controlplane kubeconfig secret %q: %w", client.ObjectKeyFromObject(src), err)
-	}
-	cfg, err := clientcmd.RESTConfigFromKubeConfig(src.Data["kubeconfig"])
-	if err != nil {
-		return fmt.Errorf("failed to deserialize the kubeconfig: %w", err)
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(secret), secret); err != nil {
+		return fmt.Errorf("failed to get controlplane service account signing key %q: %w", client.ObjectKeyFromObject(secret), err)
 	}
-	cfg.Host = strings.Replace(cfg.Host, "kube-apiserver", fmt.Sprintf("kube-apiserver.%s.svc", hcp.Namespace), 1)
-	// Impersonate the serviceaccounts groups as a guardrail to bugs in our code
-	// as we end up writing this into a public location.
-	// The perms for this are bound through the system:service-account-issuer-discovery
-	// clusterrolebinding which is managed by the KAS itself.
-	cfg.Impersonate.Groups = []string{"system:serviceaccounts"}
-	cfg.Impersonate.UserName = "iHaveToBeSubmittedButDontActuallyExist"
 
-	transport, err := rest.TransportFor(cfg)
-	if err != nil {
-		return fmt.Errorf("failed to construct transport for cfg: %w", err)
+	if !hasKeys(secret, serviceSignerPublicKey) {
+		return fmt.Errorf("controlplane service account signing key secret %q missing required key %s", client.ObjectKeyFromObject(secret), serviceSignerPublicKey)
 	}
-	client := &http.Client{Transport: transport}
-
-	for _, path := range oidcPublicDocumentPaths() {
-		requestURL := cfg.Host + path
-		req, err := http.NewRequestWithContext(ctx, http.MethodGet, requestURL, nil)
-		if err != nil {
-			return fmt.Errorf("failed to construct request for %s: %w", requestURL, err)
-		}
-
-		resp, err := client.Do(req)
-		if err != nil {
-			return fmt.Errorf("failed to do request for %s: %w", requestURL, err)
-		}
-		defer resp.Body.Close()
 
-		if resp.StatusCode != http.StatusOK {
-			body, _ := ioutil.ReadAll(resp.Body)
-			return fmt.Errorf("request to %s returned with unexpected status code %d. Response body: %s", requestURL, resp.StatusCode, string(body))
-		}
+	params := oidcGeneratorParams{
+		issuerURL: hcp.Spec.IssuerURL,
+		pubKey:    secret.Data[serviceSignerPublicKey],
```

Comment:
Might want to extract this into a local and verify the key exists so that if it does not exist we can throw a meaningful error

Reply:
`hasKeys` verifies that the key exists before ref it here